lib/nodejs/lib/thrift/xhr_connection.js

```diff
@@ -102,13 +103,22 @@ XHRConnection.prototype.flush = function() {
 
   var xreq = this.getXmlHttpRequestObject();
 
+  if (this.protocol == TBinaryProtocol) {
```

Comment:
which binary protocol is this referring to?
databricks thrift uses Http Thrift protocol not Binary thrift protocol. Is this referring to the same conectp?

Reply:
Ya, each connection within thrift has an associated protocol for encoding as well as a transport type. We use an HTTP connection within the nodeconnector, and we encode that information with a binary protocol and send it in chunks (bufffered transport).

Reply:
For the sake of [terminology](https://thrift.apache.org/docs/concepts.html): HTTP is a **transport** which can of course be combined with a binary **protocol**.